File: Add Code Here/JAVA/krishnamurthynumber.java

import java.util.Scanner;

class krishnamurthynumber {

    static int fact(int number) {
        int f = 1;
        while (number != 0) {
            f = f * number;
            number--;
        }
        return f;
    }

    static boolean checkNumber(int number) {
        int sum = 0;

        int tempNumber = number;
        while (tempNumber != 0) {

            sum = sum + fact(tempNumber % 10);

            tempNumber = tempNumber / 10;
        }

        if (sum == number)
            return true;
        else
            return false;
    }

    public static void main(String[] args) {
        int n;
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter any number:");
        s n = sc.nextInt();

        if (checkNumber(n))
            System.out.println(n + " is a krishnamurthy number");
        else
            System.out.println(n + "is not a krishnamurthy number");
    }
}

File: Add Code Here/C/Krishnamurthy Number.c
#include<stdio.h>
int main()
{
	int n,temp,fact,sum=0,i,rem;
	printf("ENTER A NUMBER: ");
	scanf("%d",&n);
	temp=n;
	while(n!=0)
	{
		rem=n%10;
		fact=1;
		for(i=rem;i>=1;i--)
  		{	
   			fact=fact*i;
  		}
  		sum=sum+fact;
  		n=n/10;
 	}
 	if(sum==temp)
  		printf("%d IS A KRISHNAMURTY NUMBER\n",temp);
 	else
  		printf("%d IS NOT A KRISHNAMURTY NUMBER\n",temp);
 	return 0;
}